ompi/mpi/c/comm_join.c

```diff
@@ -74,6 +78,10 @@ int MPI_Comm_join(int fd, MPI_Comm *intercomm)
         }
     }
 
+    if (!ompi_mpi_dynamics_is_enabled(FUNC_NAME)) {
+        return OMPI_ERRHANDLER_INVOKE(MPI_COMM_WORLD, MPI_ERR_SPAWN, FUNC_NAME);
+    }
+
```

Comment:
Similarly to the others process management functions please raise OMPI_ERR_NOT_SUPPORTED.